tests/kernel/gen_isr_table/Kconfig

```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2020 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+source "Kconfig.zephyr"
+
+config TEST_NUM_IRQS
+	int
+	# On nRF9160 and application core in nRF5340, not all interrupts with
+	# highest numbers are implemented. Thus, limit the number of interrupts
+	# reported to the test, so that it does not try to use some unavailable
+	# ones.
+	default 33 if SOC_NRF5340_CPUAPP || SOC_NRF9160
+	default NUM_IRQS
```

Comment:
Style: given that this is completely internal to the test, is it really clearer to put this all the way up here in a kconfig vs. just having this in the C code:

```c
#if defined(CONFIG_SOC_NRF5340_CPUAPP) || defined(CONFIG_SOC_NRF9160)
#define TEST_NUM_IRQS 33
#else 
#define TEST_NUM_IRQS NUM_IRQS
#endif 
```

Reply:
@andyross Thank you for reviewing and adding this comment. Indeed, handling of these specific cases entirely in the C code is clearer, and eliminates the need of having this dirty workaround of defining the `NUM_IRQS` option for architectures that don't use it. I think I focused too much on adding a Kconfig option after reading https://github.com/zephyrproject-rtos/zephyr/pull/25405#discussion_r428483278.
I refactored this PR. Would you mind taking one more look?